angular/src/app/shared/data-table/data-table.component.ts

```diff
@@ -71,6 +65,10 @@ export class DataTableComponent implements OnDestroy {
     this.selectedRowsChanged.emit(this.gridApi.getSelectedRows());
   }
 
+  onRowClicked(params: CellClickedEvent): void {
+    this.rowClicked.emit(params.node.data);
```

Comment:
Instead of emitting new event -> `this.onSelectionChanged()` (you will need to make `params?` optional in `onSelectionChanged`.